contracts/token/TransferManager.sol

```diff
@@ -138,13 +107,33 @@ contract TransferManager is Pausable, ERC20 {
         require(!frozen[_to] && !frozen[msg.sender]);
         require(_value <= balanceOf(msg.sender).sub(frozenTokens[msg.sender]), "Insufficient Balance");
         if (identityRegistry.isVerified(_to) && compliance.canTransfer(msg.sender, _to, _value)) {
-            updateShareholders(_to);
-            pruneShareholders(msg.sender, _value);
+            compliance.transferred(msg.sender, _to, _value);
             return super.transfer(_to, _value);
         }
 
         revert("Transfer not possible");
     }
+
+    /**
+     * @dev Called by an agent to pause, triggers stopped state.
+     */
+    function pause() public onlyAgent whenNotPaused {
+        _paused = true;
+        emit Paused(msg.sender);
+    }
+
+    /**
+     * @dev Called by an agent to unpause, returns to normal state.
+     */
+    function unpause() public onlyAgent whenPaused {
+        _paused = false;
+        emit UnPaused(msg.sender);
+    }
+
+    function getIdentityRegistry() public view returns (IIdentityRegistry) {
```

Comment:
Maybe rename it as a getter `identityRegistry()`? Otherwise you don't event need it, as the state variable is declared as public here https://github.com/TokenySolutions/T-REX/pull/47/files#diff-7d97aa2a3edbb863ef5e5c6a47a7a22bR20

Reply:
you need it as the state varable is not accessible from interfaces and as we only import interfaces in external contracts, we could change the getters into `identityRegistry()` and `compliance()` though but i don't really think it's a necessity

Reply:
and the state variables have been updated to `private` in the next PR

Reply:
Me are more used to have getter in the form "identityRegistry" instead of "getIdentityRegistry" (take example on ERCx interfaces, you don't have getBalanceOf() or getTokenSupply(), you have balanceOf() and tokenSupply())